cpp/src/parquet/arrow/schema_internal.cc

```diff
@@ -200,7 +217,7 @@ Result<std::shared_ptr<ArrowType>> GetArrowType(
     case ParquetType::DOUBLE:
       return ::arrow::float64();
     case ParquetType::BYTE_ARRAY:
-      return FromByteArray(logical_type);
+      return use_binary_large_variant ? FromLargeByteArray(logical_type) : FromByteArray(logical_type);
```

Comment:
You may directly pass `use_binary_large_variant` to `FromByteArray` and avoid duplicate code.